File: app/src/main/java/com/vmenon/mpo/activity/LibraryActivity.java
package com.vmenon.mpo.activity;

import android.arch.lifecycle.Observer;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.widget.DividerItemDecoration;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import com.vmenon.mpo.R;
import com.vmenon.mpo.adapter.EpisodesAdapter;
import com.vmenon.mpo.adapter.LibraryAdapter;
import com.vmenon.mpo.api.Episode;
import com.vmenon.mpo.core.persistence.MPORepository;

import org.parceler.Parcels;

import java.util.List;

import javax.inject.Inject;

public class LibraryActivity extends BaseDrawerActivity implements EpisodesAdapter.EpisodeSelectedListener {

    @Inject
    protected MPORepository mpoRepository;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getAppComponent().inject(this);

        final RecyclerView libraryList = findViewById(R.id.libraryList);
        LinearLayoutManager layoutManager = new LinearLayoutManager(this);
        libraryList.setHasFixedSize(true);
        libraryList.setLayoutManager(layoutManager);
        libraryList.addItemDecoration(new DividerItemDecoration(this, DividerItemDecoration.VERTICAL));

        mpoRepository.getAllEpisodes().observe(this, new Observer<List<Episode>>() {
            @Override
            public void onChanged(@Nullable List<Episode> episodes) {
                LibraryAdapter adapter = new LibraryAdapter(episodes);
                adapter.setListener(LibraryActivity.this);
                libraryList.setAdapter(adapter);
            }
        });
    }

    @Override
    public void onEpisodeSelected(Episode episode) {
        Intent intent = new Intent(this, EpisodeDetailsActivity.class);
        intent.putExtra(EpisodeDetailsActivity.EXTRA_EPISODE, Parcels.wrap(episode));
        startActivity(intent);
    }

    @Override
    protected int getLayoutResourceId() {
        return R.layout.activity_library;
    }

    @Override
    protected int getNavMenuId() {
        return R.id.nav_library;
    }

    @Override
    protected boolean isRootActivity() {
        return true;
    }
}

File: app/src/main/java/com/vmenon/mpo/activity/LibraryActivity.kt
package com.vmenon.mpo.activity

import androidx.lifecycle.Observer
import android.content.Intent
import android.os.Bundle
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager

import com.vmenon.mpo.R
import com.vmenon.mpo.adapter.LibraryAdapter
import com.vmenon.mpo.model.EpisodeModel
import com.vmenon.mpo.core.persistence.MPORepository
import kotlinx.android.synthetic.main.activity_library.*

import javax.inject.Inject

class LibraryActivity : BaseDrawerActivity(), LibraryAdapter.LibrarySelectedListener {

    @Inject
    lateinit var mpoRepository: MPORepository

    override val layoutResourceId: Int
        get() = R.layout.activity_library

    override val navMenuId: Int
        get() = R.id.nav_library

    override val isRootActivity: Boolean
        get() = true

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        appComponent.inject(this)

        val layoutManager = LinearLayoutManager(this)
        libraryList.setHasFixedSize(true)
        libraryList.layoutManager = layoutManager
        libraryList.addItemDecoration(DividerItemDecoration(this, DividerItemDecoration.VERTICAL))

        mpoRepository.allEpisodes.observe(this, Observer { episodes ->
            val adapter = LibraryAdapter(episodes)
            adapter.setListener(this@LibraryActivity)
            libraryList.adapter = adapter
        })
    }

    override fun onEpisodeSelected(episode: EpisodeModel) {
        val intent = Intent(this, EpisodeDetailsActivity::class.java)
        intent.putExtra(EpisodeDetailsActivity.EXTRA_EPISODE, episode.id)
        intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TASK
        startActivity(intent)
    }
}